plugins/ccgx-dmc/fu-ccgx-dmc-device.c

```diff
@@ -690,39 +692,39 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 	manual_replug =
 	    fu_device_has_private_flag(device, FU_CCGX_DMC_DEVICE_FLAG_HAS_MANUAL_REPLUG);
 
-	if (fu_device_get_update_state(self) != FWUPD_UPDATE_STATE_SUCCESS)
-		return TRUE;
-
-	if (manual_replug) {
-		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
-		fu_device_add_problem(device, FWUPD_DEVICE_PROBLEM_UPDATE_PENDING);
-	} else {
-		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
-	}
-
+	/* device action required */
 	if (self->update_model == FU_CCGX_DMC_UPDATE_MODEL_DOWNLOAD_TRIGGER) {
 		if (self->trigger_code > 0) {
 			if (!fu_ccgx_dmc_device_send_download_trigger(self,
 								      self->trigger_code,
 								      error)) {
-				if (manual_replug == FALSE) {
-					fu_device_remove_flag(device,
-							      FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
-				}
 				g_prefix_error(error, "download trigger error: ");
 				return FALSE;
 			}
 		}
 	} else if (self->update_model == FU_CCGX_DMC_UPDATE_MODEL_PENDING_RESET) {
-		if (!fu_ccgx_dmc_device_send_sort_reset(self, manual_replug, error)) {
-			if (manual_replug == FALSE) {
-				fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
-			}
+		if (!fu_ccgx_dmc_device_send_soft_reset(self, manual_replug, error)) {
 			g_prefix_error(error, "soft reset error: ");
 			return FALSE;
 		}
 	}
 
+	/* the user has to do something */
+	if (manual_replug) {
+		g_autoptr(FwupdRequest) request = fwupd_request_new();
+		fwupd_request_set_kind(request, FWUPD_REQUEST_KIND_IMMEDIATE);
+		fwupd_request_set_id(request, FWUPD_REQUEST_ID_REMOVE_REPLUG);
+		fwupd_request_add_flag(request, FWUPD_REQUEST_FLAG_ALLOW_GENERIC_MESSAGE);
+		fwupd_request_set_message(
+		    request,
+		    "The update will continue when the device USB cable has been "
```

Comment:
We've got this string in like 3 or 4 plugins now, can we make it generic?

Reply:
It literally is `FWUPD_REQUEST_FLAG_ALLOW_GENERIC_MESSAGE` :) Or do you mean "if we set FWUPD_REQUEST_ID_REMOVE_REPLUG then automatically use this text"?

Reply:
I mean the string literally. Like can we have a define for it?

Reply:
Ahh, I think we can do better than that -- how about the follow-up commit?